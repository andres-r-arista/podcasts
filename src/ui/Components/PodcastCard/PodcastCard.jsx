import React from "react";

import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";

export const PodcastCard = ({ title, author, imgUrl, id }) => {
  const navigate = useNavigate();
  return (
    <div className="w-1/2 md:w-1/4" data-testid="podcast-card">
      <div className="border text-center h-24 mx-4 my-16 shadow-md cursor-pointer" data-testid="podcast-card-box" onClick={() => navigate(`/podcast/${id}`)}>
        <div className="flex flex-col -translate-y-8 items-center">
          <img src={imgUrl} alt={title} data-testid="podcast-card-image" className="w-16 h-16 rounded-full" />
          <h1
            className="uppercase font-semibold truncate max-w-full text-sm px-2"
            data-testid="podcast-card-title"
            title={title}
          >
            {title}
          </h1>
          <h2
            className="text-gray-400 font-semibold truncate max-w-full text-xs px-2"
            data-testid="podcast-card-author"
            title={author}
          >{`Author: ${author}`}</h2>
        </div>
      </div>
    </div>
  );
};

PodcastCard.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
