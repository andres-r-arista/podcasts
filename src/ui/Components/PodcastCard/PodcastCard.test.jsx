import { render, screen, fireEvent } from "@testing-library/react";

import { PodcastCard } from "./PodcastCard";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

const title = "Title";
const author = "Author";
const imgUrl = "http://www.google.com";
const id = "123";

describe("Tests in <PodcastCard />", () => {
  test("must match the snapshot", () => {
    const { container } = render(
      <PodcastCard title={title} author={author} imgUrl={imgUrl} id={id} />
    );
    expect(container).toMatchSnapshot();
  });

  test("must display the correct image url", () => {
    render(
      <PodcastCard title={title} author={author} imgUrl={imgUrl} id={id} />
    );
    const screenImage = screen.getByTestId("podcast-card-image");
    expect(screenImage.src).toContain("http://www.google.com");
  });

  test("must display the correct title", () => {
    render(
      <PodcastCard title={title} author={author} imgUrl={imgUrl} id={id} />
    );
    const screenTitle = screen.getByTestId("podcast-card-title");
    expect(screenTitle).toHaveTextContent("Title");
  });

  test("must display the correct author", () => {
    render(
      <PodcastCard title={title} author={author} imgUrl={imgUrl} id={id} />
    );
    const screenAuthor = screen.getByTestId("podcast-card-author");
    expect(screenAuthor).toHaveTextContent(`Author: ${author}`);
  });

  test("must navigate when the box is clicked", async () => {
    render(
      <PodcastCard title={title} author={author} imgUrl={imgUrl} id={id} />
    );
    const box = screen.getByTestId("podcast-card-box");
    await expect(mockedUsedNavigate).toHaveBeenCalledTimes(0);
    fireEvent.click(box);
    await expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    await expect(mockedUsedNavigate).toHaveBeenCalledWith(`/podcast/${id}`);
  });
});
