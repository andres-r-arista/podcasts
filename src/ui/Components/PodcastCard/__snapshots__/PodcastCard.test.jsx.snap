// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PodcastCard /> must match the snapshot 1`] = `
<div>
  <div
    class="w-1/2 md:w-1/4"
    data-testid="podcast-card"
  >
    <div
      class="border text-center h-24 mx-4 my-16 shadow-md cursor-pointer"
      data-testid="podcast-card-box"
    >
      <div
        class="flex flex-col -translate-y-8 items-center"
      >
        <img
          alt="Title"
          class="w-16 h-16 rounded-full"
          data-testid="podcast-card-image"
          src="http://www.google.com"
        />
        <h1
          class="uppercase font-semibold truncate max-w-full text-sm px-2"
          data-testid="podcast-card-title"
          title="Title"
        >
          Title
        </h1>
        <h2
          class="text-gray-400 font-semibold truncate max-w-full text-xs px-2"
          data-testid="podcast-card-author"
          title="Author"
        >
          Author: Author
        </h2>
      </div>
    </div>
  </div>
</div>
`;
